projects/observability/src/shared/components/cartesian/d3/chart/cartesian-chart.ts

```diff
@@ -65,14 +68,35 @@ export class DefaultCartesianChart<TData> implements CartesianChart<TData> {
     onEvent: ChartEventListener<TData>;
   }[] = [];
 
+  protected readonly brushBehaviour: BrushBehavior<unknown>;
+
   public constructor(
     protected readonly hostElement: Element,
     protected readonly injector: Injector,
     protected readonly renderingStrategy: RenderingStrategy,
     protected readonly svgUtilService: SvgUtilService,
     protected readonly d3Utils: D3UtilService,
     protected readonly domRenderer: Renderer2
-  ) {}
+  ) {
+    this.brushBehaviour = brushX<unknown>().on('end', () => this.onBrushSelection(_d3CurrentEvent));
```

Comment:
Please only set this up if the interaction handler is provided by parent component.

Reply:
done